### PR TITLE
Adding json schema validation for ontology generation

### PR DESCRIPTION
**Pull request by mgupta1410**

### CODE REVIEW

File: forte/data/ontology/configs/example_multi_module_ontology_config.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    {
+      "entry_name": "custom.user.custom_module.Dependency",
+      "parent_entry": "forte.data.ontology.top.Link",
+      "attributes": [
+        {
+          "name": "rel_type",
+          "type": "str"
+        }
+      ],
+      "class_attributes": [
```
Comment (hunterhector) [inline forte/data/ontology/configs/example_multi_module_ontology_config.json]: Still class_attributes here.

File: forte/data/ontology/configs/example_ontology_config.json [hunk 2]
```diff
@@ -1,25 +1,51 @@
...
+                    "type": "List",
+                    "item_type": "ft.onto.example_ontology.Word"
+                }
+            ]
+        },
+        {
+            "entry_name": "ft.onto.example_ontology.WordLink",
+            "parent_entry": "Link",
+            "description": "",
+            "class_attributes": [
```
Comment (hunterhector) [inline forte/data/ontology/configs/example_ontology_config.json]: class_attributes

File: forte/data/ontology/configs/stanfordnlp_ontology.json [hunk 3]
```diff
@@ -1,58 +1,62 @@
...
+      "entry_name": "ft.onto.stanfordnlp_ontology.Document",
       "parent_entry": "forte.data.ontology.top.Annotation"
     },
     {
-      "entry_name": "forte.data.ontology.stanfordnlp_ontology.Dependency",
+      "entry_name": "ft.onto.stanfordnlp_ontology.Dependency",
       "parent_entry": "forte.data.ontology.top.Link",
-      "parent_type": "forte.data.ontology.stanfordnlp_ontology.Token",
-      "child_type": "forte.data.ontology.stanfordnlp_ontology.Token",
+      "class_attributes": [
```
Comment (hunterhector) [inline forte/data/ontology/configs/stanfordnlp_ontology.json]: class_attributes

File: .travis.yml [hunk 4]
```diff
@@ -11,7 +11,7 @@ env:
 install:
   - pip install --upgrade pip
   - pip install --progress-bar off torch==$TORCH_VER
-  - pip install --progress-bar off .[ner,srl,txtgen,stanfordnlp,test,example,indexer]
+  - pip install --progress-bar off .[ner,srl,txtgen,stanfordnlp,test,example,indexer,codegen]
```
Comment (hunterhector) [inline .travis.yml]: codegen requirements would be part of core, so we don't need to use an extra keyword here.

File: forte/data/ontology/utils.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+"""
+    Utility functions for ontology generation.
+"""
+import json
+import jsonschema
+
+
+def validate_json_schema(input_filepath: str, validation_filepath: str):
+    """
```
Comment (hunterhector) [inline forte/data/ontology/utils.py]: Let's mention that what exception will be thrown once it failed.

File: forte/data/ontology/test/test_configs/test_additional_properties.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+                {
+                    "name": "pos",
+                    "type": "str"
+                },
+                {
+                    "name": "pos",
+                    "type": "str"
+                }
+            ],
+            "class_attributes": [
```
Comment (hunterhector) [inline forte/data/ontology/test/test_configs/test_additional_properties.json]: Is this test intended to show that ``class_attributes`` are not allowed?

**Comment by mgupta1410** [comment]
Yes, it is to make sure that the `additionalProperties=false` (
https://json-schema.org/understanding-json-schema/reference/object.html#properties)
of the JSON validation schema is working as intended.

File: forte/data/ontology/test/test_configs/test_additional_properties.json (hunk 6 above)
Comment (mgupta1410) [reply forte/data/ontology/test/test_configs/test_additional_properties.json]: Yes, it is to make sure that the `additionalProperties=false` (https://json-schema.org/understanding-json-schema/reference/object.html#properties) of the JSON validation schema is working as intended.

Comment (mgupta1410) [reply forte/data/ontology/test/test_configs/test_additional_properties.json]: https://github.com/asyml/forte/pull/95/files#diff-01ba68ac9bab2f1e5d7083dedaadeaa0R42 - This test makes sure that an error is thrown if there are `class_attributes` in the schema, as the property is not allowed.

**Pull request merged**